### PR TITLE
fix swipe-refresh spinner showing forever when refreshing AccountActivity

### PR DESCRIPTION
**Pull request by connyduck**
The flow must emit every update even if the values are the same, so use SharedFlow instead of StateFlow.

Regression from https://github.com/tuskyapp/Tusky/pull/4337 cc @Goooler

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/account/AccountViewModel.kt [hunk 1]
```diff
@@ -42,8 +45,8 @@ class AccountViewModel @Inject constructor(
     private val _noteSaved = MutableStateFlow(false)
     val noteSaved: StateFlow<Boolean> = _noteSaved.asStateFlow()
 
-    private val _isRefreshing = MutableStateFlow(false)
-    val isRefreshing: StateFlow<Boolean> = _isRefreshing.asStateFlow()
+    private val _isRefreshing = MutableSharedFlow<Boolean>(1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+    val isRefreshing: SharedFlow<Boolean> = _isRefreshing
```
Comment (Goooler) [inline app/src/main/java/com/keylesspalace/tusky/components/account/AccountViewModel.kt:49]:
```suggestion
    val isRefreshing: SharedFlow<Boolean> = _isRefreshing.asSharedFlow()
```

Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/components/account/AccountViewModel.kt:49]: Need `import kotlinx.coroutines.flow.asSharedFlow`

**Review by Goooler — APPROVED** [review]

**Review by Tak — APPROVED** [review]

**Comment by cbeyls** [comment]
A better fix would be to set `isRefreshing` to true first in the coroutine that sets `isRefreshing` to false at the end.
This would also allow to properly show the refresh indicator after a configuration change if loading is still in progress.

**Comment by connyduck** [comment]
Yes, I just wanted a quick fix to get rid of the regression. There are probably more improvements possible in the account view, I want to have a look soon. PRs also welcome!

**Comment by Goooler** [comment]
Fixing in #4353.

**Pull request merged**